app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -138,19 +161,23 @@ public void success(PagedList<Event> eventsPagedList) {
                         List<Event> events = eventsPagedList.getItems();
                         Timber.d("Got events");
                         if (events.size() > 0) {
-                            Event firstEvent = events.get(0);
-                            views.setTextViewText(R.id.title, firstEvent.getTitle());
-                            views.setTextViewText(R.id.location, firstEvent.getLocation());
-                            views.setTextViewText(R.id.startDate,
-                                firstEvent.getStart().toLocalDateTime()
-                                    .toString(DateTimeFormat.patternForStyle("MS",
-                                        getResources().getConfiguration().locale)));
-                            showChild(views, 1);
-
-                            Intent i = new Intent(UpdateService.this, EventActivity.class);
-                            i.putExtra(Const.EXTRA_EVENT_ID, firstEvent.getId());
-                            views.setOnClickPendingIntent(R.id.container,
-                                PendingIntent.getActivity(UpdateService.this, 0, i, 0));
+                            Event firstEvent = getNextEvent(events);
```

Comment:
Sorry for the late comment here, it is not that important but it would be great if we change the variable to `nextEvent`. It is not the first event anymore.